rust/operator-binary/src/main.rs

```diff
@@ -0,0 +1,120 @@
+mod controller;
+
+use clap::Parser;
+use futures::stream::StreamExt;
+use stackable_operator::cli::Command;
+use stackable_operator::{
+    k8s_openapi::api::{
+        apps::v1::StatefulSet,
+        core::v1::{ConfigMap, Endpoints, Service},
+    },
+    kube::{
+        api::{DynamicObject, ListParams},
+        runtime::{
+            controller::{Context, ReconcilerAction},
+            reflector::ObjectRef,
+            Controller,
+        },
+        CustomResourceExt, Resource,
+    },
+};
+use stackable_trino_crd::TrinoCluster;
+
+mod built_info {
+    include!(concat!(env!("OUT_DIR"), "/built.rs"));
+}
+
+#[derive(Parser)]
+#[clap(about = built_info::PKG_DESCRIPTION, author = stackable_operator::cli::AUTHOR)]
+struct Opts {
+    #[clap(subcommand)]
+    cmd: Command,
+}
+
+/// Erases the concrete types of the controller result, so that we can merge the streams of multiple controllers for different resources.
+///
+/// In particular, we convert `ObjectRef<K>` into `ObjectRef<DynamicObject>` (which carries `K`'s metadata at runtime instead), and
+/// `E` into the trait object `anyhow::Error`.
+fn erase_controller_result_type<K: Resource, E: std::error::Error + Send + Sync + 'static>(
+    res: Result<(ObjectRef<K>, ReconcilerAction), E>,
+) -> anyhow::Result<(ObjectRef<DynamicObject>, ReconcilerAction)> {
+    let (obj_ref, action) = res?;
+    Ok((obj_ref.erase(), action))
+}
+
+#[tokio::main]
+async fn main() -> anyhow::Result<()> {
+    stackable_operator::logging::initialize_logging("TRINO_OPERATOR_LOG");
+
+    let opts = Opts::parse();
+    match opts.cmd {
+        Command::Crd => println!("{}", serde_yaml::to_string(&TrinoCluster::crd())?,),
+        Command::Run(product_config) => {
+            stackable_operator::utils::print_startup_string(
+                built_info::PKG_DESCRIPTION,
+                built_info::PKG_VERSION,
+                built_info::GIT_VERSION,
+                built_info::TARGET,
+                built_info::BUILT_TIME_UTC,
+                built_info::RUSTC_VERSION,
+            );
+            let product_config = product_config.product_config.load(&[
+                "deploy/config-spec/properties.yaml",
+                "/etc/stackable/trino-operator/config-spec/properties.yaml",
+            ])?;
+
+            let client =
+                stackable_operator::client::create_client(Some("trino.stackable.tech".to_string()))
+                    .await?;
+
+            let trino_controller_builder =
+                Controller::new(client.get_all_api::<TrinoCluster>(), ListParams::default());
+            let trino_store = trino_controller_builder.store();
+
+            let trino_controller = trino_controller_builder
+                .owns(client.get_all_api::<Service>(), ListParams::default())
```

Comment:
Not sure, but this looks like it'll own *all* Service objects. I saw that it is like that in other operators too, so it'll probably not be a problem, but maybe someone can explain this to me?

Reply:
It just means all namespaces vs `client.get_namespaced_api(...)`